### ISSUE TITLE
org-roam-db-build-cache doesn't properly handle encrypted notes

### ISSUE THREAD

**Issue by spolakh** [org-roam/org-roam#1194]
### Description

I've been using org-roam over a collection of encrypted org notes `(setq org-roam-encrypt-files t)` for a couple of months now. It was going pretty smooth, until recently when in order to try to debug unrelated issues I've removed my `org-roam.db` file. After that two things broke: 
1. Each note now is referred to by the truncated filename (`my_note.gpg` - notably not `my_note.gpg.org` which is an actual filename) instead of the title (`My Note`). By referred I mean the way it is named in the lookups (`org-roam-find-file`) and in the first line of org-roam sidebar.
2. org-roam sidebar now shows `* No backlinks!` for each note

Note1: everything works well for the new files that I created after deleting the db  

#### Steps to Reproduce

1. Load Emacs with a following config:
```
(setq org-roam-encrypt-files t)
(setq epa-file-encrypt-to "yourgpgkeyname@example.com")
(setq org-roam-directory "~/tmp/org-roam-test")
(setq org-roam-db-location "~/tmp/org-roam-test.db")
(add-hook 'after-init-hook 'org-roam-mode)
(require 'org-roam)
```

2. Run `org-roam-find-file` to create "My First Note"
3. Run `org-roam-insert` to create "My Second Note" and link to it from the first one
4. Exit emacs
5. Run `rm ~/tmp/org-roam-test.db`
6. Open emacs
7. Run `org-roam-db-build-cache`
8. Run `org-roam-find-file` to open "My Second Note" - notice that it's referred to by the filename not "My Second Note"
9. Observe that when opened sidebar shows "* No backlinks!"
10. You can wipe the `~/tmp/org-roam-test` folder and try 1.-9. with `(setq org-roam-encrypt-files nil)` and observe that steps 8. and 9. give different results (you can see "My Second Note" in the find minibuffer and opened sidebar shows a reference from "My First Note") 

### Environment

```
Copy info below this line into issue:
- Emacs: GNU Emacs 27.1 (build 1, x86_64-apple-darwin19.5.0, NS appkit-1894.50 Version 10.15.5 (Build 19F101))
 of 2020-08-27
- Framework: Doom
- Org: Org mode version 9.4 (nil @ /Users/spolakh/.emacs.d/.local/straight/build/org-mode/)
- Org-roam: 1.2.2
```

Note2: The fix might be somewhat tricky to implement? I imagine going through all the encrypted files during each `org-roam-db-build-cache` will take forever unless we skip files that haven't been modified since the last time it ran? (maybe that's how `org-roam-db-build-cache` already operates, not sure? - my elisp-fu is pretty weak so far to get through the code) 

Also, any ideas about how to force-index my existing notes until the fix for this gets released, are super-welcome - I'm currently pretty lost in trying to come up with a hotfix

**Comment by spolakh** [comment]
As suggested by @jethrokuan on Slack, running `org-roam-db-build-cache` with a universal argument (`C-u M-x org-roam-db-build-cache` with Emacs keybindings, `SPC u M-x org-roam-db-build-cache` with evil keybindings) correctly force re-indexes all encrypted files

**Comment by jethrokuan** [comment]
To expand on what was explained above:

Org-roam does not do anything special to support encrypted files, these are all using Emacs in-built capabilities. To build the cache, Emacs calls `insert-file-contents` on the GPG-encrypted file. Here, the observed behaviour is from Emacs being unable to decrypt the file.

When calling `C-u M-x org-roam-db-build-cache`, it performs the the same steps described above: it deletes the database file, and recreates the database. The key difference here is that Emacs has been started already (for a while), and it seems to make a difference for whether a decryption will be successful/performed or not.

TL;DR seems to be a real issue, but not something that's fixable with Org-roam.
